api/provider/aws/service_update.go

```diff
@@ -0,0 +1,71 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (s *ServiceProvider) Update(req models.UpdateServiceRequest) error {
+	serviceID := req.ServiceID
+
+	environmentID, err := lookupEntityEnvironmentID(s.TagStore, "service", serviceID)
+	if err != nil {
+		return err
+	}
+
+	fqEnvironmentID := addLayer0Prefix(s.Config.Instance(), environmentID)
+	clusterName := fqEnvironmentID
+
+	fqServiceID := addLayer0Prefix(s.Config.Instance(), serviceID)
+	serviceName := fqServiceID
+
+	if req.DeployID != nil {
+		taskDefinitionID := *req.DeployID
+		if err := s.updateServiceTaskDefinition(clusterName, serviceName, taskDefinitionID); err != nil {
```

Comment:
The [docs](https://github.com/aws/aws-sdk-go/blob/master/service/ecs/api.go#L8927) show that the `TaskDefinition` needs to be in `family:revision` format. With our new deploy implementation, Layer0 deploy ids are just randomly generated strings; their `family` and `revision` properties are stored as tags, under `name` and `version`, respectively. 
This needs an extra step converting a layer0 deploy ID to the format `family:revision` - or, perhaps even better, we always pass around those 2 arguments are to separate things, e.g.:
```
 updateServiceTaskDefinition(clusterName, serviceName, taskDefinitionFamily string, taskDefinitionRevision int)
```

Reply:
✅